analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SillyMathematicalComparison.CSharp9.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace SonarAnalyzer.UnitTest.TestCases
+{
+    internal class SillyMathematicalComparison
+    {
+        public void IsPatterns()
+        {
+            _ = 33 is 55; // Compliant
```

Comment:
Would be useful to have a pattern here that would be `// Noncompliant {{Comparison to this constant is useless; the constant is outside the range of type 'float'}}`. I assume that such a pattern will not compile. That way, we have documented that we don't have to care about it for the rule.

This case is actually an FN: It should read `// Noncompliant {{Don't compare constant values}}`. Please mark it as such (No need to fix it).

Reply:
I am not sure what you mean about the first case you want me to add. Something like this?

```csharp
const double d = double.MaxValue;
float x = 42;
_ = x is d;
```